### ISSUE TITLE
[Bug]: When the app is brought back to Foreground, false events are triggered continuously

### ISSUE THREAD

**Issue by KamathVishak96** [OutdatedGuy/internet_connection_checker_plus#105]
### Is there an existing issue for this?

- [x] I have searched the existing issues

### Bug Description

When I bring the app to the Foreground, the listener receives lots of events together in this sequence - disconnected, connected, disconnected, connected, disconnected, connected, ... so on. The frequency is very short, like close to 10 milliseconds. And this happens on a very stable connectivity and happens across both iOS and Android Devices, across different networks.

As suggested in another issue, we have implemented the pause() and resume() when we noticed this issue. Still, it doesn't change anything. I can confirm that the await InternetConnection().onStatusChange.listen() is called only once. Also, I am calling the listener only after 1st API call completes. 

### Steps to Reproduce

1. Add a small delay after the page opens and listen to the stream, pause and resume on relevant lifecycle events
2. Put the app in the background, and leave it for a minute
3. Bring the app to the foreground
4. Observe the Logs

### Expected Behavior

When the app is brought to the foreground and the listener is resumed, it should not emit if there is no connectivity change

### Actual Behavior

The disconnect and connect states are emitted continuously and sequentially, even when on a Stable network.

### Package Version

2.9.1

### Platform(s)

Android, iOS

### Flutter Channel

stable

### Flutter Doctor Output

```shell
[✓] Flutter (Channel stable, 3.32.3, on macOS 15.3.2 24D81 darwin-arm64, locale en-IN) [1,667ms]
    • Flutter version 3.32.3 on channel stable at /Users/vishakakamath/development/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 5c1433509f (5 months ago), 2025-06-11 13:12:24 -0700
    • Engine revision 31c4875c7a
    • Dart version 3.8.1
    • DevTools version 2.45.1

[✓] Android toolchain - develop for Android devices (Android SDK version 35.0.0) [3.6s]
    • Android SDK at /Users/vishakakamath/Library/Android/sdk
    • Platform android-36, build-tools 35.0.0
    • ANDROID_HOME = /Users/vishakakamath/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jbr/Contents/Home/bin/java
      This is the JDK bundled with the latest Android Studio installation on this machine.
      To manually set the JDK path, use: `flutter config --jdk-dir="path/to/jdk"`.
    • Java version OpenJDK Runtime Environment (build 21.0.5+-13047016-b750.29)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 16.4) [2.9s]
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 16F6
    • CocoaPods version 1.16.2

[✓] Chrome - develop for the web [10ms]
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2024.3) [10ms]
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 21.0.5+-13047016-b750.29)

[✓] VS Code (version 1.98.2) [9ms]
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension can be installed from:
      🔨 https://marketplace.visualstudio.com/items?itemName=Dart-Code.flutter

[✓] Connected device (3 available) [6.3s]
    • WHITE iPhone XR (mobile) • 00008020-000C50DE0180402E • ios            • iOS 16.6 20G75
    • macOS (desktop)          • macos                     • darwin-arm64   • macOS 15.3.2 24D81 darwin-arm64
    • Chrome (web)             • chrome                    • web-javascript • Google Chrome 142.0.7444.176

[✓] Network resources [464ms]
    • All expected network resources are available.

• No issues found!
```

### Minimal Reproduction Code

```dart
@override
void onInit() {
    readUser()
}


void readUser() async {
    // API Call here
    listenInternetConnectionChanges();
}

void listenInternetConnectionChanges() async {
    if (isListenedToInternet) return;
    isListenedToInternet = true;
    internetConnectionSubscription =
        await InternetConnection().onStatusChange.listen((status) {
         Sentry.addBreadcrumb(
          Breadcrumb(message: "Connectivity changed: $status"));
         
        if (status == InternetStatus.connected) {
         // Some code
        } else {
         // Some code
        }
    }
}
```

### Relevant Log Output

```shell
Here's a Screenshot of Logs in Sentry. There is no exceptions.
<img width="917" height="811" alt="Image" src="https://github.com/user-attachments/assets/681b7b28-ec9a-4abb-8d1b-b348a899dae3" />
```

### Device/Network Information

```markdown

```

### Would you like to work on this issue?

- [ ] I'm interested in working on this issue

**Comment by patrikjohnson-elisa** [comment]
@KamathVishak96 Not sure if this is your issue, but using `await` for the continuous listener like in your example code does not make sense conceptually and might have something to do with it...

**Comment by KamathVishak96** [comment]
@patrikjohnson-elisa Apologies, that await doesn't exist in our codebase added I just typed in the code and added it by mistake here. But the code you see here is the same way we're able to reproduce the issue, of course, except for the `await`.

Just want to add our findings over the last few days. When I put the app in the Background, I am pausing the Listener, and on resume, I resume it. When the app is in the Background, and I switch Internet On/Off, let's say three times, then when I come back to the app, I see three sets of connect-disconnect events. If I do it 10 times, I'd see 10 sets of events.
